packages/kolme/src/core/types/error.rs

```diff
@@ -19,6 +25,227 @@ pub enum KolmeError {
+        found: BridgeActionId,
+    },
+
+    #[error("Validator {signer} already approved proposal {proposal_id}")]
+    AlreadyApprovedProposal {
+        signer: PublicKey,
+        proposal_id: AdminProposalId,
+    },
+
+    #[error("Conflicting block in DB at height {height} with hash {hash}")]
+    ConflictingBlockInDb { height: u64, hash: Sha256Hash },
+
+    #[error("Failed to execute signed Cosmos bridge transaction: {details}")]
+    CosmosExecutionFailed { details: String },
+
+    #[error("Timed out proposing and awaiting transaction {txhash}: {details}")]
+    TimeoutProposingTx { txhash: TxHash, details: String },
+
+    #[error("API server error: {details}")]
+    ApiServerError { details: String },
+
+    #[error("Mismatched genesis info: actual {actual:?}, expected {expected:?}")]
+    MismatchedGenesisInfo {
+        actual: GenesisInfo,
+        expected: GenesisInfo,
+    },
+
+    #[error("Identical proposal {id} already exists")]
+    DuplicateAdminProposal { id: AdminProposalId },
+
+    #[error("Invalid signature: expected signer {expected}, actual {actual}")]
+    InvalidSignature {
+        expected: Box<PublicKey>,
+        actual: Box<PublicKey>,
+    },
+
+    #[error("Executed block height mismatch: expected {expected}, got {actual}")]
+    ExecutedHeightMismatch {
+        expected: BlockHeight,
+        actual: BlockHeight,
+    },
+
+    #[error("Submitter error: {0}")]
+    Submitter(#[from] SubmitterError),
+
+    #[error("Import/export error: {0}")]
+    ImportExport(#[from] KolmeImportExportError),
+
+    #[error("Types error: {0}")]
+    TypesError(#[from] KolmeTypesError),
+
+    #[error("Core error: {0}")]
+    CoreError(#[from] KolmeCoreError),
+
+    #[error("Listener error: {0}")]
+    ListenerError(#[from] KolmeListenerError),
+
+    #[error("Execution error: {0}")]
+    ExecuteError(#[from] KolmeExecuteError),
+
+    #[error("Execution error: {0}")]
+    Execution(#[from] KolmeExecutionError),
+
+    #[error("Store error: {0}")]
+    StoreError(#[from] KolmeStoreError),
+
+    #[error("Failed to serialize Solana payload to Borsh: {details}")]
+    SolanaPayloadSerializationError { details: String },
+
+    #[error("Failed to build Solana initialization transaction: {details}")]
+    SolanaInitTxBuildFailed { details: String },
+
+    #[error("Solana submitter failed to execute signed transaction: {details}")]
+    SolanaSignedTxExecutionFailed { details: String },
+
+    #[error("Failed to create Solana pubsub client: {0}")]
+    SolanaPubsubError(String),
+
+    #[error("Bridge program {program} hasn't been initialized yet")]
+    UninitializedSolanaBridge { program: String },
+
+    #[error("Error deserializing Solana bridge state: {details}")]
+    InvalidSolanaBridgeState { details: String },
+
+    #[error("Error deserializing Solana bridge message from logs: {details}")]
+    InvalidSolanaBridgeLogMessage { details: String },
+
     #[error("{0}")]
     Other(String),
```

Comment:
Is it possible to remove the `Other` variant? I only see it used in the `impl_from_to_other!` macro and the `From<anyhow::Error>` impl below.  Ideally all of these cases can be made into specific variants that embed the source Error.